src/trpc/router/restRouter.ts

```diff
@@ -205,6 +238,215 @@ export const restRouter = router({
+                registrationEmail: z.string(),
+                corporateEmail: z.string(),
+                active: z.boolean(),
+                supervisorLogin: z.string().nullish(),
+            }),
+        )
+        .query(async ({ input }) => {
+            const user = await userMethods.getByLogin(input.login);
+            let newSupervisor: Awaited<ReturnType<typeof userMethods.getByLogin>> | null = null;
+
+            if (user.supervisor?.login !== input.data.supervisorLogin) {
+                newSupervisor = await userMethods.getByLogin(input.data.supervisorLogin);
+            }
+
+            const fieldsToUpdate: Record<string, string> = {};
+
+            if (user.organizationUnitId !== input.data.organizationUnitId) {
+                const newOrganization = await prisma.organizationUnit.findUnique({
+                    where: {
+                        id: input.data.organizationUnitId,
+                    },
+                });
+
+                if (newOrganization) fieldsToUpdate.organizationUnitId = newOrganization.id;
+            }
+
+            const updatedName = trimAndJoin([input.data.surname, input.data.firstName, input.data.middleName]);
+
+            if (updatedName !== user.name) fieldsToUpdate.name = updatedName;
+            if (newSupervisor?.id && newSupervisor?.id !== user.supervisorId) {
+                fieldsToUpdate.supervisorId = newSupervisor.id;
+            }
+            if (input.data.registrationEmail !== user.email) fieldsToUpdate.email = input.data.registrationEmail;
+
+            const { supervisorId, email, ...rest } = await prisma.user.update({
+                where: { id: user.id },
+                data: fieldsToUpdate,
+            });
+
+            const phoneService = await prisma.userService.findFirst({
+                where: {
+                    userId: user.id,
+                    serviceName: 'Phone',
+                },
+            });
+
+            if (phoneService?.serviceName !== input.data.phone) {
+                if (phoneService) {
+                    await prisma.userService.update({
+                        where: {
+                            serviceName_serviceId: {
+                                serviceName: 'Phone',
+                                serviceId: phoneService.serviceId,
+                            },
+                        },
+                        data: {
+                            serviceName: 'Phone',
+                            serviceId: input.data.phone,
+                        },
+                    });
+                } else {
+                    await prisma.userService.create({
+                        data: {
+                            userId: user.id,
+                            serviceName: 'Phone',
+                            serviceId: input.data.phone,
+                        },
+                    });
+                }
+            }
+
+            return {
```

Comment:
Should create "edit user" history event before return

Reply:
fixed